src/sys/windows/selector.rs

```diff
@@ -192,11 +161,12 @@ impl SockState {
             self.poll_info.handles[0].status = 0;
             self.poll_info.handles[0].events = self.user_evts | afd::POLL_LOCAL_CLOSE;
 
-            let wrapped_overlapped = OverlappedArcWrapper::new(self_arc);
-            let overlapped = wrapped_overlapped.get_ptr() as *const _ as PVOID;
             let result = unsafe {
-                self.afd
-                    .poll(&mut self.poll_info, (*self.iosb).as_mut_ptr(), overlapped)
+                self.afd.poll(
+                    &mut self.poll_info,
+                    &mut *self.iosb,
+                    transmute_copy(self_arc),
```

Comment:
Can you move this `transmute_copy` out the call and use explicit types, transmute is generally a very unsafe function.